### ISSUE TITLE
Parser deadlock

### ISSUE THREAD

**Issue by dharmab** [dharmab/skyeye#369]
```
Nov 02 03:58:31 skyeye.eu.limakilo.net systemd[1]: Stopping skyeye.service - SkyEye GCI Bot...
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: 3:58AM WRN shutdown took too long, forcing exit
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: goroutine profile: total 7
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: 1 @ 0x41b129 0x47fc29 0x527113 0x485fe1
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x47fc28        os/signal.signal_recv+0x28        /opt/hostedtoolcache/go/1.23.0/x64/src/runtime/>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x527112        os/signal.loop+0x12                /opt/hostedtoolcache/go/1.23.0/x64/src/os/sign>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: 1 @ 0x43e951 0x47cc7d 0x550d71 0x550ba5 0x54d9cb 0xb1a3ee 0x485fe1
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x550d70        runtime/pprof.writeRuntimeProfile+0xb0        /opt/hostedtoolcache/go/1.23.0/x64/>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x550ba4        runtime/pprof.writeGoroutine+0x44        /opt/hostedtoolcache/go/1.23.0/x64/src/r>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x54d9ca        runtime/pprof.(*Profile).WriteTo+0x14a        /opt/hostedtoolcache/go/1.23.0/x64/>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb1a3ed        main.run.func1+0x8d                        /home/runner/work/skyeye/skyeye/cmd/sk>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: 1 @ 0x47dd4e 0x4149ad 0x414617 0xa09825 0xa02df3 0x485fe1
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xa09824        github.com/dharmab/skyeye/pkg/simpleradio.(*client).decodeVoice+0x4a4        /hom>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xa02df2        github.com/dharmab/skyeye/pkg/simpleradio.(*client).Run.func5+0x52        /home/r>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: 1 @ 0x47dd4e 0x4149ad 0x414617 0xb166fb 0xb15ce5 0xb13313 0x485fe1
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb166fa        github.com/dharmab/skyeye/internal/application.(*app).recognizeSample+0x8da      >
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb15ce4        github.com/dharmab/skyeye/internal/application.(*app).recognize+0x1c4            >
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb13312        github.com/dharmab/skyeye/internal/application.(*app).Run.func5+0x52             >
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: 1 @ 0x47dd4e 0x45c365 0x45c334 0x47f0e5 0x4945e8 0xb1a268 0x666131 0x666a1f 0xb18cda 0xb18ccf 0x4494ab 0x4>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x47f0e4        sync.runtime_Semacquire+0x24                                /opt/hostedtoolcache/>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x4945e7        sync.(*WaitGroup).Wait+0x47                                /opt/hostedtoolcache/g>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb1a267        main.run+0x747                                                /home/runner/work/s>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x666130        github.com/spf13/cobra.(*Command).execute+0xa90                /home/runner/go/pk>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x666a1e        github.com/spf13/cobra.(*Command).ExecuteC+0x3fe        /home/runner/go/pkg/mod/g>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb18cd9        github.com/spf13/cobra.(*Command).Execute+0x59                /home/runner/go/pkg>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb18cce        main.main+0x4e                                                /home/runner/work/s>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x4494aa        runtime.main+0x28a                                        /opt/hostedtoolcache/go>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: 1 @ 0x47dd4e 0x45c365 0x45c334 0x47f0e5 0x4945e8 0xb1a339 0x485fe1
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x47f0e4        sync.runtime_Semacquire+0x24        /opt/hostedtoolcache/go/1.23.0/x64/src/runtim>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0x4945e7        sync.(*WaitGroup).Wait+0x47        /opt/hostedtoolcache/go/1.23.0/x64/src/sync/wa>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb1a338        main.run.func2+0xb8                /home/runner/work/skyeye/skyeye/cmd/skyeye/mai>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: 1 @ 0x4809c5 0xa32039 0xa3203a 0xa34855 0xb158ad 0xb154a5 0xb12e37 0x485fe1
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xa32038        bufio.(*Scanner).Text+0x98                                                /opt/ho>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xa32039        github.com/dharmab/skyeye/pkg/parser.(*parser).parseDeclare+0x99        /home/run>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xa34854        github.com/dharmab/skyeye/pkg/parser.(*parser).Parse+0x14b4                /home/>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb158ac        github.com/dharmab/skyeye/internal/application.(*app).parseText+0x30c        /hom>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb154a4        github.com/dharmab/skyeye/internal/application.(*app).parse+0x64        /home/run>
Nov 02 03:58:41 skyeye.eu.limakilo.net skyeye[404560]: #        0xb12e36        github.com/dharmab/skyeye/internal/application.(*app).Run.func8+0x56        /home>
N
```

**Comment by dharmab** [comment]
Potential fix in #370

**Comment by dharmab** [comment]
Looks fixed to me
